### PR TITLE
Task refactoring

### PR DESCRIPTION
**Pull request by artemredkin**
Instead of having promise and a future inside Handler and Task, I propose to hide the promise inside the Task and drop future from it

### CODE REVIEW

File: Sources/NIOHTTPClient/HTTPHandler.swift [hunk 1]
```diff
@@ -272,8 +276,16 @@ public extension HTTPClient {
             }
         }
 
+        public func succeed(_ value: Response) {
```
Comment (ianpartridge) [inline Sources/NIOHTTPClient/HTTPHandler.swift]: Does this need to be `public`? Should a user of the library be able to succeed/fail the task themselves?

File: Sources/NIOHTTPClient/HTTPHandler.swift [hunk 2]
```diff
@@ -236,22 +236,26 @@ internal extension URL {
...
-        init(eventLoop: EventLoop, future: EventLoopFuture<Response>) {
+        public init(eventLoop: EventLoop) {
             self.eventLoop = eventLoop
-            self.future = future
+            self.promise = eventLoop.makePromise()
             self.cancelled = false
             self.lock = Lock()
         }
 
+        public var futureResult: EventLoopFuture<Response> {
```
Comment (tanner0101) [inline Sources/NIOHTTPClient/HTTPHandler.swift]: I think if `promise` is public, we probably don't need this or the succeed, wait, fail, etc methods.

File: Sources/NIOHTTPClient/HTTPHandler.swift (hunk 1 above)
Comment (artemredkin) [reply Sources/NIOHTTPClient/HTTPHandler.swift]: good point, this shouldn't be public, thanks!

File: Sources/NIOHTTPClient/HTTPHandler.swift (hunk 2 above)
Comment (artemredkin) [reply Sources/NIOHTTPClient/HTTPHandler.swift]: actually, as Ian pointed out, it's better to have promise internal, so users cannot accidentally succeed it. Other methods are just convenience helpers

**Pull request merged**